velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -107,63 +107,111 @@ void ensureRegexIsConstantAndCompatible(
 // If position <= 0, throw error.
 // If position > length string, return string.
 template <typename T>
-struct RegexReplaceFunction {
+struct RegexpReplaceFunction {
   VELOX_DEFINE_FUNCTION_TYPES(T);
 
-  void call(
+  bool callNullable(
       out_type<Varchar>& result,
-      const arg_type<Varchar>& stringInput,
-      const arg_type<Varchar>& pattern,
-      const arg_type<Varchar>& replace) {
-    re2::RE2* patternRegex = getCachedRegex(pattern.str());
-    re2::StringPiece replaceStringPiece = toStringPiece(replace);
+      const arg_type<Varchar>* stringInput,
+      const arg_type<Varchar>* pattern,
+      const arg_type<Varchar>* replace) {
+    if (!stringInput) {
+      result.resize(0);
+      return false;
+    } else if (!pattern || !replace) {
+      result.resize(stringInput->size());
+      std::memcpy(result.data(), stringInput->data(), stringInput->size());
+      return true;
+    }
+    re2::RE2* patternRegex = nullptr;
+    try {
+      patternRegex = getCachedRegex(pattern->str());
+    } catch (const VeloxUserError& e) {
+      result.resize(0);
+      throw e;
+    }
+    re2::StringPiece replaceStringPiece = toStringPiece(*replace);
 
-    std::string string(stringInput.data(), stringInput.size());
+    std::string string(stringInput->data(), stringInput->size());
     RE2::GlobalReplace(&string, *patternRegex, replaceStringPiece);
 
     if (string.size()) {
       result.resize(string.size());
       std::memcpy(result.data(), string.data(), string.size());
+      return true;
     } else {
       result.resize(0);
+      return false;
     }
   }
 
-  void call(
+  bool callNullable(
       out_type<Varchar>& result,
-      const arg_type<Varchar>& stringInput,
-      const arg_type<Varchar>& pattern,
-      const arg_type<Varchar>& replace,
-      const arg_type<int64_t>& position) {
-    VELOX_USER_CHECK_GE(position, 1, "regex_replace requires a position >= 1");
+      const arg_type<Varchar>* stringInput,
+      const arg_type<Varchar>* pattern,
+      const arg_type<Varchar>* replace,
+      const arg_type<int64_t>* position) {
+    if (!position) {
+      return callNullable(result, stringInput, pattern, replace);
+    } else if (!stringInput) {
+      result.resize(0);
+      return false;
+    } else if (!pattern || !replace) {
+      result.resize(stringInput->size());
+      std::memcpy(result.data(), stringInput->data(), stringInput->size());
+      return true;
+    }
 
-    re2::RE2* patternRegex = getCachedRegex(pattern.str());
-    re2::StringPiece replaceStringPiece = toStringPiece(replace);
-    re2::StringPiece inputStringPiece = toStringPiece(stringInput);
+    try {
+      VELOX_USER_CHECK_GT(
+          *position, 0, "regexp_replace requires a position >= 1");
+    } catch (const VeloxUserError& e) {
```

Comment:
This logic doesn't make sense to me. Why are you doing it? You should be able to use VELOX_USER_CHECK_GT as is without try-catch-rethrow.

CC: @laithsakka @kgpai